### PR TITLE
Add prebuilt ICU Analyzer

### PR DESCRIPTION
**Pull request by russcam**
Relates: #3615, elastic/elasticsearch#34958

### CODE REVIEW

File: src/Nest/Analysis/Plugins/Icu/Collation/IcuCollationCaseFirst.cs [hunk 1]
```diff
@@ -5,9 +5,12 @@
 namespace Nest
 {
 	/// <summary>
-	/// Sets the alternate handling for strength quaternary to be either shifted or non-ignorable.
-	/// Which boils down to ignoring punctuation and whitespace.
+	/// Controls which case is sorted first when case is not ignored for
```
Comment (codebrain) [inline src/Nest/Analysis/Plugins/Icu/Collation/IcuCollationCaseFirst.cs]: Quite the comment change!

Comment (russcam) [reply src/Nest/Analysis/Plugins/Icu/Collation/IcuCollationCaseFirst.cs]: The summary was incorrect; was copied from `IcuCollationAlternate`

File: src/Nest/Analysis/Plugins/Icu/Collation/IcuCollationStrength.cs [hunk 2]
```diff
@@ -9,6 +9,9 @@ namespace Nest
 	/// difference considered significant during comparison.
 	/// See also: http://icu-project.org/apiref/icu4j/com/ibm/icu/text/Collator.html
 	/// </summary>
+	/// <remarks>
```
Comment (Mpdreamz) [inline src/Nest/Analysis/Plugins/Icu/Collation/IcuCollationStrength.cs]: :+1:

File: src/Nest/Analysis/Plugins/Icu/IcuCollationTokenFilter.cs [hunk 3]
```diff
@@ -132,38 +135,38 @@ public class IcuCollationTokenFilterDescriptor
 		string IIcuCollationTokenFilter.VariableTop { get; set; }
 		string IIcuCollationTokenFilter.Variant { get; set; }
 
-		/// <inheritdoc />
+		/// <inheritdoc cref="IIcuCollationTokenFilter.Language" />
```
Comment (Mpdreamz) [inline src/Nest/Analysis/Plugins/Icu/IcuCollationTokenFilter.cs]: :100:

File: src/Tests/Tests/Analysis/Analyzers/AnalyzerTests.cs [hunk 4]
```diff
@@ -69,14 +69,14 @@ public class SimpleTests : AnalyzerAssertionBase<SimpleTests>
...
-		public class LanguageTests : AnalyzerAssertionBase<SimpleTests>
+		public class LanguageTests : AnalyzerAssertionBase<LanguageTests>
 		{
 			public override FuncTokenizer Fluent => (n, an) => an
 				.Language("myLanguage", a => a.Language(Language.Dutch));
 
 			public override IAnalyzer Initializer => new LanguageAnalyzer { Language = Language.Dutch };
 
-			public override object Json => new { type = "dutch" };
+			public override object Json => new { type = "Dutch" };
```
Comment (Mpdreamz) [inline src/Tests/Tests/Analysis/Analyzers/AnalyzerTests.cs]: There is one test failure because of this

Comment (russcam) [reply src/Tests/Tests/Analysis/Analyzers/AnalyzerTests.cs]: d'oh, thanks

**Pull request merged**